File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/06.Nested-Loop/02-exercise/04.trainTheTrainers.js
function trainTheTrainers(input) {

    const juryCount = Number(input[0]);
    let index = 1;
    let current = input[index++];

    let sumOfAllGrades = 0;
    let presentationCount = 0;

    while (current !== "Finish") {
        presentationCount++;
        let sumOfGrades = 0;
        for (let i = 1; i <= juryCount; i++) {
            const currentGrade = Number(input[index++]);
            sumOfGrades += currentGrade;
            sumOfAllGrades += currentGrade;
        }
        const averageGrade = sumOfGrades / juryCount;
        console.log(`${current} - ${averageGrade.toFixed(2)}.`);

        current = input[index++];
    }

    const averageGradeOfAll = sumOfAllGrades / (juryCount * presentationCount);
    console.log(`Student's final assessment is ${averageGradeOfAll.toFixed(2)}.`);
}
trainTheTrainers(["2",
"Objects and Classes",
"5.77",
"4.23",
"Dictionaries",
"4.62",
"5.02",
"RegEx",
"2.88",
"3.42",
"Finish"]);

File: Python_Basics/Homeworks-And-Labs/06.nested_loops/02.exercise/04.train_the_trainers.py
jury_count = int(input())
current = input()

sum_of_all_grades = 0
presentation_count = 0

while current != "Finish":
    presentation_count += 1
    sumOfGrades = 0
    for i in range(jury_count):
        current_grade = float(input())
        sumOfGrades += current_grade
        sum_of_all_grades += current_grade
        
    average_grade = sumOfGrades / jury_count
    print(f'{current} - {average_grade:.2f}.')

    current = input()

average_gradeOfAll = sum_of_all_grades / (jury_count * presentation_count)
print(f"Student's final assessment is {average_gradeOfAll:.2f}.")
